community/kernel/src/main/java/org/neo4j/kernel/impl/factory/GraphDatabaseFacade.java

```diff
@@ -862,17 +1019,26 @@ void closeResources( Statement statement )
             IOUtils.closeAllSilently( statement, nodeLabelCursor, nodeCursor, propertyCursor );
         }
 
-        private boolean hasPropertyWithValue()
+        private boolean hasPropertiesWithValues()
```

Comment:
think this looks simpler:
```
        private boolean hasPropertiesWithValues()
        {
            read.singleNode( nodeLabelCursor.nodeReference(), nodeCursor );
            if ( nodeCursor.next() )
            {
                nodeCursor.properties( propertyCursor );
                while ( propertyCursor.next() )
                {
                    for ( IndexQuery query : queries )
                    {
                        if ( propertyCursor.propertyKey() != query.propertyKeyId() || !query.acceptsValueAt( propertyCursor ) )
                        {
                            return false;
                        }
                    }
                }
                return true;
            }
            return false;
        }
```

Reply:
I don't think this is doing the same thing? The node may have properties which are not queried for, as long as it also has every queried for property. Here we would `return false` on the first non-queried for property, which would exclude nodes incorrectly.

However, you made me think more about this, and actually if we find a propertyKey matching a query, _but with the wrong value_, we can return early. Added that.